formula/src/main/java/com/instacart/formula/internal/FormulaManagerImpl.kt

```diff
@@ -245,6 +245,40 @@ internal class FormulaManagerImpl<Input, State, Output>(
         return manager.run(input).output
     }
 
+    fun <ChildInput, ChildOutput> child(
+        key: Any,
+        formula: IFormula<ChildInput, ChildOutput>,
+        input: ChildInput,
+        onError: (Throwable) -> Unit,
+    ): ChildOutput? {
+        val childrenManager = getOrInitChildrenManager()
+        val manager = childrenManager.findOrInitChild(key, formula, input)
+
+        // If termination happens while running, we might still be initializing child formulas. To
+        // ensure correct behavior, we mark each requested child manager as terminated to avoid
+        // starting new actions.
+        if (isTerminated()) {
+            manager.markAsTerminated()
+            manager.performTerminationSideEffects()
+        }
+        manager.setValidationRun(isValidationEnabled)
+
+        return try {
+            if (!manager.isTerminated()) {
```

Comment:
The preceding performTerminationSideEffects was added because we are not executing the child formula in the this condition. My reasoning is that there is a good chance this will throw another exception and continually throwing every evaluation is a performance concern.